authmodules/TYPO3/privacyidea/Classes/PrivacyideaService.php

```diff
@@ -110,23 +113,30 @@ public function init() {
 	 * @return int authentication statuscode, one of 0, 100 and 200
 	 */
 	public function authUser(array $user) {
-		// 0 means authentication failure
-		$ret = 0;
-		$username = $this->login['uname'];
-		$password = $this->login['uident_text'];
-		$this->logger->info("try to authenticate user [$username]");
+		if ($this->extConf['privacyIDEABackend'] != 'adminOnly'
+			|| (bool)$user['admin']
+			|| $this->authInfo['loginType'] == 'FE'
+		) {
```

Comment:
yeah this is really untransparent :(
I want to do this check:
```
if (                                                                                                                               
  ($this->authInfo['loginType'] == 'BE' && $this->extConf['privacyIDEABackend'] == 'adminOnly' && (bool)$user['admin']) ||             
  ($this->authInfo['loginType'] == 'BE' && $this->extConf['privacyIDEABackend'] == 'always') ||                                        
  ($this->authInfo['loginType'] == 'FE')
)
```
But in the init() function, there is already checked whether to do the auth or not so we can omit a few checks. For reasons of readability, would the upper logic be a better solution?